File: extract_docx.js
const AdmZip = require('adm-zip');
const path = require('path');

const docxPath = path.join(__dirname, 'consultation', '现代琳凯蒂亚语 v26.5.docx');
const zip = new AdmZip(docxPath);

// Extract document.xml which contains the main content
const documentXml = zip.readAsText('word/document.xml');

// Simple XML text extraction - strip tags
const text = documentXml
  .replace(/<w:p[^>]*>/g, '\n')  // paragraph start
  .replace(/<w:br[^>]*\/>/g, '\n')  // line break
  .replace(/<w:tab[^>]*\/>/g, '\t')  // tab
  .replace(/<[^>]+>/g, '')  // remove all other tags
  .replace(/&amp;/g, '&')
  .replace(/&lt;/g, '<')
  .replace(/&gt;/g, '>')
  .replace(/&quot;/g, '"')
  .replace(/&apos;/g, "'")
  .replace(/&#x/g, '&#x')  // keep hex entities for now
  .replace(/\n\s*\n/g, '\n')  // remove empty lines
  .trim();

// Write to a text file for reading
const fs = require('fs');
const outPath = path.join(__dirname, 'consultation', 'document_text.txt');
fs.writeFileSync(outPath, text, 'utf-8');
console.log('Text extracted to:', outPath);
console.log('--- First 5000 chars ---');
console.log(text.substring(0, 5000));

File: extract_docx.py
import docx
import os

doc_path = r'c:\Users\CandyGirl\Desktop\光线传奇\琳凯蒂亚语\琳凯蒂亚文化网站设计2026.06\consultation\现代琳凯蒂亚语 v26.5.docx'
doc = docx.Document(doc_path)

with open('docx_content.txt', 'w', encoding='utf-8') as f:
    for i, p in enumerate(doc.paragraphs):
        f.write(f'[{i}] {p.text}\n')

print(f'Total paragraphs: {len(doc.paragraphs)}')
print('Content saved to docx_content.txt')